src/git_provider/github.rs

```diff
@@ -12,6 +23,59 @@ impl GitHub {
 }
 
 impl GitProvider for GitHub {
+    fn set_token(&self, path: &PathBuf) -> Result<String> {
+        let token = dialoguer::Input::<String>::with_theme(&ColorfulTheme::default())
+            .with_prompt(
+                "Set up your GitHub personal access token. You can generate one at:
+https://github.com/settings/tokens/new?description=Gi&scopes=repo,read:org,read:user,user:email\n",
+            )
+            .interact()?;
+
+        let data = json!({
+            "accessToken": token
+        });
+
+        fs::write(path, data.to_string())
+            .context("Failed to write to token file")
+            .suggestion("Check if you have write permissions")?;
+
+        Ok(token)
+    }
+
+    fn get_token(&self) -> Result<String> {
+        let home_dir = dirs::home_dir().ok_or_eyre("Failed to get home directory")?;
+        let token_file = home_dir.join(".config/gi/token");
+
+        if !token_file.exists() {
+            create_dir_all(home_dir.join(".config/gi"))
+                .context("Failed to create gi config directory")
+                .suggestion("Check if you have write permissions to the .config directory")?;
+
+            File::create_new(&token_file)
+                .context("Failed to create token file")
+                .suggestion("Check if you have write permissions to the .config/gi directory")?;
+
+            let permissions = fs::Permissions::from_mode(0o600);
+            fs::set_permissions(&token_file, permissions)
+                .context("Failed to set permissions on token file")?;
```

Comment:
I'm pretty sure this won't compile on windows. Let's leave a TODO here that explains the current limitation.

```suggestion
            // TODO: explain
            let permissions = fs::Permissions::from_mode(0o600);
            fs::set_permissions(&token_file, permissions)
                .context("Failed to set permissions on token file")?;
```